prov/rxd/src/rxd_cq.c

```diff
@@ -323,6 +325,15 @@ static int rxd_verify_active(struct rxd_ep *ep, fi_addr_t addr, fi_addr_t peer_a
 	}
 
 	ep->peers[addr].peer_addr = peer_addr;
+
+	if (!dlist_empty(&ep->peers[addr].unacked)) {
+		dlist_pop_front(&ep->peers[addr].unacked,
+				struct rxd_pkt_entry, pkt_entry, d_entry);
+		dlist_remove(&pkt_entry->d_entry);
+		rxd_release_tx_pkt(ep, pkt_entry);
+		ep->peers[addr].unacked_cnt--;
+		dlist_remove(&ep->peers[addr].entry);
```

Comment:
I need help here.  We're removing stuff from 3 different lists.  The packet is stored on 2 lists?  Is this correct?

Reply:
First dlist_remove is an extra one. I'll take it out. We're removing the rts packet from the unacked list and then removing the peer from the rts_sent_list.